src/runtime_src/core/common/drv/xrt_xclbin.c

```diff
@@ -207,8 +207,13 @@ xrt_xclbin_get_section_hdr(const struct axlf *xclbin, enum axlf_section_kind kin
 		return NULL;
 
 	for (i = 0; i < xclbin->m_header.m_numSections; i++) {
-		if (xclbin->m_sections[i].m_sectionKind == kind)
+		if (xclbin->m_sections[i].m_sectionKind == kind){
+			int err = 0;
+			err = xrt_xclbin_check_section_hdr(&(xclbin->m_sections[i]), xclbin->m_header.m_length);
```

Comment:
Is `xrt_xclbin_check_section_hdr` secure?  I think not, there can be overflow in the arithmetic such that the sum is less that `xclbin_len`, right?

I am reviewing the secure coding training we have been assigned.  And it has a good (not intuitive) example of how to deal with arithmetic overflow when checking for size.   

The check line 226 has to be strengthen to something like this:
`return ( (offset + size) < offset) || (offset + size) > xclbin_len) ? -EINVAL : 0;`

Reply:
Thank you @stsoe, made suggested change.